crates/ruma-common/src/events/room/member.rs

```diff
@@ -377,45 +395,126 @@ fn membership_change(
 }
 
 impl OriginalRoomMemberEvent {
+    /// Obtain the details about this event that are required to calculate a membership change.
+    ///
+    /// This is required when you want to calculate the change a redacted `m.room.member` event
+    /// made.
+    pub fn details(&self) -> MembershipDetails<'_> {
+        self.content.details()
+    }
+
+    fn prev_details(&self) -> Option<MembershipDetails<'_>> {
```

Comment:
Hmm, it probably makes sense to make this pub as well. A client will want to print out that the display name changed from -> to. This method sounds like it could be useful for this.

Reply:
Then we would have to make the fields public. I think I'd rather add that stuff to `MembershipChange`.

Reply:
I'll create a separate issue for this.